bot/__init__.py

```diff
@@ -101,131 +101,118 @@ def parse_python(view_pos):
             if self.buffer.endswith("]"):
+
                 # Key: The first argument, specified `bot.tags[here]`
-                key = self.buffer[self.index + 1:self.buffer.rfind("]")]
+                key = tempbuffer[self.index + 1:tempbuffer.rfind("]")]
                 log.trace(f"Command mimicks getitem. Key: {key!r}")
-
-                # note: if not key, this corresponds to an empty argument
-                #       so this should throw / return a SyntaxError ?
                 args = ast.literal_eval(key)
 
-                # Use the cog's `get` command.
-                python_result = self.buffer[self.previous:self.index] + ".get"
+                # Use the cogs `.get` method.
+                parsed_result = self.buffer[self.previous:self.index] + ".get"
 
             # Syntax is `bot.tags['ask'] = 'whatever'` => mimic `setattr`
             elif "=" in self.buffer and not self.buffer.endswith("="):
-                equals_pos = self.buffer.find("=")
-                closing_bracket_pos = self.buffer.rfind("]", 0, equals_pos)
+                equals_pos = tempbuffer.find("=")
+                closing_bracket_pos = tempbuffer.rfind("]", 0, equals_pos)
 
                 # Key: The first argument, specified `bot.tags[here]`
-                key_contents = self.buffer[self.index + 1:closing_bracket_pos]
+                key_contents = tempbuffer[self.index + 1:closing_bracket_pos]
                 key = ast.literal_eval(key_contents)
 
                 # Value: The second argument, specified after the `=`
-                right_hand = self.buffer.split("=", maxsplit=1)[1].strip()
+                right_hand = tempbuffer.split("=", maxsplit=1)[1].strip()
                 value = ast.literal_eval(right_hand)
 
                 # If the value is a falsy value - mimick `bot.tags.delete(key)`
                 if not value:
                     log.trace(f"Command mimicks delitem. Key: {key!r}.")
-                    python_result = self.buffer[self.previous:self.index] + ".delete"
+                    parsed_result = self.buffer[self.previous:self.index] + ".delete"
                     args = key
 
                 # Otherwise, assume assignment, for example `bot.tags['this'] = 'that'`
                 else:
-                    # Allow using double quotes in triple double quote string
-                    value = value.replace('"', '\\"')
                     log.trace(f"Command mimicks setitem. Key: {key!r}, value: {value!r}.")
-
-                    # Use the cog's `set` command.
-                    python_result = self.buffer[self.previous:self.index] + ".set"
-                    args = f'"{key}" "{value}"'
+                    parsed_result = self.buffer[self.previous:self.index] + ".set"
+                    args = (key, value)
 
             # Syntax is god knows what, pass it along
-            # in the future, this should probably return / throw SyntaxError
```

Comment:
so we scrapped the `SyntaxError` altogether as its currently not possible to implement, right?